app/Http/Controllers/CategoryController.php

```diff
@@ -13,42 +13,52 @@ function index(){
         return view('admin.category.index',['categories'=>$categories]);
     }
 
-    function add(){
+    function create(){
         return view('admin.category.add');
     }
 
-    function postAdd(Request $request){
-        $category = new Category();
-        $category->name = $request->name;
-        $category->description = $request->description;
-        $category->save();
+    function store(Request $request){
+       $request->validate([
+            'name' => 'required|max:225',
+            'description'=> 'required'
+        ]);
+        Category::create(
+            array(
+                'name'=>$request->name,
+                'description'=>$request->description
+            )
+        );
 
         $categories = Category::all();
         return view('admin.category.index',['categories'=>$categories]);
```

Comment:
chỗ này khi tạo/sửa thành công thì mình sẽ ko return về view nửa. Mà dùng redirect để chuyển về lại page index 

[bấm đây](https://laravel.com/docs/8.x/routing#generating-urls-to-named-routes) xem docs

Cụ thể chỗ này sẽ đổi lại

```
return redirect()->route('admin.category.index');
```